### ISSUE TITLE
Job resource parameters can only be specified as a flat list of `<param>` XML elements

### ISSUE THREAD

**Issue by simonbray** [galaxyproject/galaxy#16406]
For some tools on our server, I'd like to give the user the choice of sending the tools to different destinations. This is possible specifying a parameter in the job_resource_params_conf

```
<parameters>
    <param label="Destination" name="destination" type="select" value="option1" help="Destination where job should be executed">
        <option value="option1">option1</option>
        <option value="option2">option2</option>
    </param>
</parameters>
```

and writing a dynamic job rule

```
    if param_dict['__job_resource']['destination'] == 'option1':
        return JobDestination(runner="option1")
    elif param_dict['__job_resource']['destination'] == 'option2':
        return JobDestination(runner="option2")
```

However, while this works fine, there are additional resource parameters which may or may not be relevant, depending which of the two options is picked, so I'd like to wrap the parameter in a conditional and offer separate choices of parameters depending which is picked. Unfortunately, this isn't supported currently, the code just iterates through the XML and ignores everything which isn't a `<param>` element: https://github.com/galaxyproject/galaxy/blob/dd869c61b87ef96f652cb2626076eea4c405947a/lib/galaxy/util/__init__.py#L1004-L1017

Maybe this isn't something to prioritise - probably most people don't trust their users this much - but it would be nice for the resource parameter parsing to be a bit more sophisticated. Even raising a `NotImplementedError` so it's clear this is not currently supported would be something.

**Comment by mvdbeek** [comment]
Yeah, we don't want anyone to pick things like "arbitrarily named resource 1 that is definitely not available anywhere else". I hope medium term we'll replace this with user-specified resource requests that may or may not be satisfiable, but that are at least portable.

**Comment by simonbray** [comment]
> Yeah, we don't want anyone to pick things like "arbitrarily named resource 1 that is definitely not available anywhere else". I hope medium term we'll replace this with user-specified resource requests that may or may not be satisfiable, but that are at least portable.

I understand your point (especially after seeing your PR #16579), but I think portability between servers is a separate issue. Given that job resource parameters currently exist and are useful (for us at least, and they're mentioned in the GTN) it would be good if they could work as well as possible.

**Comment by mvdbeek** [comment]
Can we take a step back and see why you want to offer the destination selection to users ? You've got quite some options to tailor execution based on input parameters, and you could even do meta scheduling with TPV. When is the user in a position to make a better selection than what you could set up as a rule ?

**Comment by simonbray** [comment]
We're assuming that our users are experienced and trustworthy, know the difference between the resources and would be able to use them via the command line anyway if they weren't using Galaxy, and prefer to be able to control things themselves where reasonable. So it doesn't make sense to codify a set of rules - we just offer a default and the option to deviate from this if useful.

We don't have to continue this discussion btw. I can fully understand you are not motivated to put work into developing a feature you don't feel should exist.

**Comment by jmchilton** [comment]
I absolutely think this feature should exist generally but conditionals are implemented in a way I find to be too complicated to advise working on this feature. Maybe if someday we are handling tool state in a more... cohesive way this can be revisited.
